var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -228,54 +228,55 @@ def install(self, spec, prefix):
 
         # take step-40 which can use both PETSc and Trilinos
         # FIXME: switch step-40 to MPI run
-        with working_dir('examples/step-40'):
-            print('=====================================')
-            print('========== Step-40 PETSc ============')
-            print('=====================================')
-            # list the number of cycles to speed up
-            filter_file(r'(const unsigned int n_cycles = 8;)',
-                        ('const unsigned int n_cycles = 2;'), 'step-40.cc')
-            cmake('.')
-            if '^petsc' in spec:
-                make('release')
-                make('run', parallel=False)
-
-            print('=====================================')
-            print('========= Step-40 Trilinos ==========')
-            print('=====================================')
-            # change Linear Algebra to Trilinos
-            filter_file(r'(\/\/ #define FORCE_USE_OF_TRILINOS.*)',
-                        ('#define FORCE_USE_OF_TRILINOS'), 'step-40.cc')
-            if '^trilinos+hypre' in spec:
-                make('release')
-                make('run', parallel=False)
+        if spec.satisfies('@8.4.0:'):
+            with working_dir('examples/step-40'):
+                print('=====================================')
+                print('========== Step-40 PETSc ============')
+                print('=====================================')
+                # list the number of cycles to speed up
+                filter_file(r'(const unsigned int n_cycles = 8;)',
+                            ('const unsigned int n_cycles = 2;'), 'step-40.cc')
+                cmake('.')
+                if '^petsc' in spec:
+                    make('release')
+                    make('run', parallel=False)
+            
+                print('=====================================')
+                print('========= Step-40 Trilinos ==========')
+                print('=====================================')
+                # change Linear Algebra to Trilinos
+                filter_file(r'(\/\/ #define FORCE_USE_OF_TRILINOS.*)',
+                            ('#define FORCE_USE_OF_TRILINOS'), 'step-40.cc')
+                if '^trilinos+hypre' in spec:
+                    make('release')
+                    make('run', parallel=False)
 
-            print('=====================================')
-            print('=== Step-40 Trilinos SuperluDist ====')
-            print('=====================================')
-            # change to direct solvers
-            filter_file(r'(LA::SolverCG solver\(solver_control\);)',  ('TrilinosWrappers::SolverDirect::AdditionalData data(false,"Amesos_Superludist"); TrilinosWrappers::SolverDirect solver(solver_control,data);'), 'step-40.cc')  # NOQA: ignore=E501
```

Comment:
btw, part of the unit test in deal.II is compilation of all examples, so it would strange that the native Step-40 did not compile on 8.3.0.